### PR TITLE
Add tests for Pandoc reader

### PR DESCRIPTION
**Pull request by nandac**

### CODE REVIEW

**Comment by nandac** [comment]
Final commit for now.

I am not happy with the way isort sorted the imports for `text_pandoc_reader.py` as it put a local import pandoc_reader above the pelican imports. Kindly let me know if I should tweak the isort settings to get this to work correctly.

I was wondering if it is okay to change the markdown heading style in the README to use `#` and `##` instead of `=====` or `-----` style since I am using multi-level headings and want to be consistent.

File: pelican/plugins/pandoc_reader/test_pandoc_reader.py [hunk 1]
```diff
@@ -0,0 +1,1157 @@
+"""Tests for pandoc-reader plugin."""
+# pylint: disable=too-many-lines
```
Comment (justinmayer) [inline pelican/plugins/pandoc_reader/test_pandoc_reader.py:2]: If `pylint` warns about too many lines, and GitHub won't even render the diff without users explicitly tapping a button to load the diff… then maybe this file does indeed have too many lines? Do you think it make make sense to split this into separate files, perhaps one file per test class?

**Comment by justinmayer** [comment]
Regarding the import sorting, I agree that it needs adjusting. I suggest the following:

```toml
# Designate "pelican" and "pandoc_reader" as separate import sections
known_pelican = "pelican"
known_pandoc = "pandoc_reader"

sections = "FUTURE,STDLIB,THIRDPARTY,PELICAN,PANDOC,FIRSTPARTY,LOCALFOLDER"
```

Regarding the Markdown heading style, I spend a lot of time looking at plain-text Markdown (i.e., not rendered), and I usually find the underlined headers _much_ more "scannable" and readable than `#` / `##`. Personally, I don't think there's anything wrong with mixing the two styles, using underlines for major headings 1 & 2 and then `###` / `####` for minor headings. But I don't feel particularly strongly about this, so you are free to change them if it's important to you.

**Comment by nandac** [comment]
Thanks @justinmayer. I have separated out the tests which definitely makes them a lot more manageable.

I have also fixed the import order with your suggestions and it works well.

I am not too worried about heading styles in Markdown myself. It is just that my vscode Markdown linter complains about mixing the two styles which I can switch off on my end.

I think the only thing left is figuring out under what name we want to publish the plugin. The existing Pelican-Pandoc-Reader in pyPI is significantly different from mine with respect to implementation and features.

**Comment by justinmayer** [comment]
Everything here looks good to me. Well done! 🥇 

Regarding the package name, I already opened an issue in this repo in hopes of merging our efforts and utilizing the existing package name for the next release. Let's allow for some time for a potential response then decide what to do next.

**Pull request merged**